assets/js/blocks-registry/payment-methods/payment-method-config-helper.js

```diff
@@ -8,3 +8,39 @@ export const canMakePaymentWithFeaturesCheck = ( canMakePayment, features ) => (
 	);
 	return featuresSupportRequirements && canMakePayment( canPayArgument );
 };
+
+// Filter out payment methods by callbacks registered by extensions
+export const canMakePaymentWithExtensions = (
+	canMakePayment,
+	extensionsCallbacks,
+	paymentMethodName
+) => ( canPayArgument ) => {
+	// validate whether the payment method is available
+	let canPay = canMakePayment( canPayArgument );
+
+	if ( canPay ) {
+		const namespacedCallbacks = {};
+		Object.entries( extensionsCallbacks ).forEach(
+			( [ namespace, callbacks ] ) => {
+				if ( typeof callbacks[ paymentMethodName ] === 'function' ) {
+					namespacedCallbacks[ namespace ] =
+						callbacks[ paymentMethodName ];
+				}
+			}
+		);
+		canPay = Object.keys( namespacedCallbacks ).every( ( namespace ) => {
+			try {
+				return namespacedCallbacks[ namespace ]( canPayArgument );
+			} catch ( err ) {
+				// eslint-disable-next-line no-console
+				console.error(
+					`Error when executing callback for ${ paymentMethodName } in ${ namespace }`,
+					err
+				);
+				throw err;
```

Comment:
We don't need to throw here as well. We need to continue with the calculation and ignore whatever result the callback would have had if it wouldn't have thrown.